cmd/admin-kms-key-list.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2015-2023 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	json "github.com/minio/colorjson"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/pkg/console"
+)
+
+var adminKMSKeyListCmd = cli.Command{
+	Name:         "list",
+	Usage:        "request list of KMS master keys",
+	Action:       mainAdminKMSKeyList,
+	OnUsageError: onUsageError,
+	Before:       setGlobalsFromContext,
+	Flags:        globalFlags,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Get list of master keys from a MinIO server/cluster.
+     $ {{.HelpName}} play
+`,
+}
+
+// adminKMSKeyCmd is the handle for the "mc admin kms key" command.
+func mainAdminKMSKeyList(ctx *cli.Context) error {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 1 {
+		showCommandHelpAndExit(ctx, 1) // last argument is exit code
+	}
+
+	console.SetColor("KeyName", color.New(color.FgBlue))
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	// Create a new MinIO Admin Client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Unable to initialize admin connection.")
+
+	keys, e := client.ListKeys(globalContext, "*")
+	fatalIf(probe.NewError(e).Trace(args...), "Unable to list KMS keys")
+
+	for _, k := range keys {
+		printMsg(kmsKeyMsg{
+			Key: k.Name,
+		})
+	}
+	return nil
```

Comment:
We should actually print all the keys together @shtripat in a table form not one key per line.

We should also print when the key was created there is lot more information in kes.KeyInfo that can be pretty printed here.

It also looks like our `madmin-go` is behind what kes-go package can provide so we must update madin-go also to comply with the kes-go changes.

Reply:
Ok, will check and send PR for `madmin-go` and use the updated info here for displaying.

Reply:
I agree about the table format.

Unfortunately, listing keys does not return any CreatedAt data, key algorithm etc.
These fields are not set by current stateless KES servers since they would have to fetch each key from the backend KMS to send that information to the client - which is expensive.

Instead the API (stateful and stateless) KES servers expose will be: https://github.com/minio/kes/blob/854064f68ab89c895d2eea9d03f41829bfba08fd/internal/api/types.go#L99

So, listing - as of now - will only return the name of the item (key, policy, secret, ...)
In the future this might be expended but for now, only the key names are returned.

Reply:
That is fine then - let's put it into a table format for now.